sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -138,16 +138,26 @@ class DetermineTableStats(session: SparkSession) extends Rule[LogicalPlan] {
     relation.copy(tableStats = stats)
   }
 
-  override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
```

Comment:
shall we simply make this rule explicitly handle subqueries? I feel we may hit the same issue if dataframe API also supports subqueries and we may have an already resolved subquery in the plan.

Reply:
Yes, I like your idea better too, but I haven't figured out how to implement it. If handling this in `ResolveSubquery#resolveSubQueries` would it cause analyzer rules to be applied multiple times.

Reply:
We can use `plan.transformWithSubqueries`

Reply:
Oh, actually we can update `ResolveSubquery#resolveSubQueries` to check `subquery.analyzed`, instead of `subquery.resolved`. The rationale is, we should always run the post-hoc resolution rules, even if the plan is already resolved.

Reply:
Thanks @cloud-fan , I will try it.